proof-producer/bin/proof-producer/CMakeLists.txt

```diff
@@ -75,14 +83,32 @@ function(setup_proof_generator_target)
         Boost::program_options
         Boost::thread
     )
+
+    target_link_libraries(${ARG_TARGET_NAME}-lib INTERFACE
+        crypto3::all
+
+        crypto3::transpiler
+
+        Boost::filesystem
+        Boost::log
+        Boost::program_options
+        Boost::thread
+    )
 endfunction()
 
 # Declare single-threaded target
 set(SINGLE_THREADED_TARGET "${CURRENT_PROJECT_NAME}-single-threaded")
 setup_proof_generator_target(TARGET_NAME ${SINGLE_THREADED_TARGET})
 
+
 # Declare multi-threaded target
 set(MULTI_THREADED_TARGET "${CURRENT_PROJECT_NAME}-multi-threaded")
 setup_proof_generator_target(TARGET_NAME ${MULTI_THREADED_TARGET} ADDITIONAL_DEPENDENCIES actor::zk)
 
 install(TARGETS ${SINGLE_THREADED_TARGET} ${MULTI_THREADED_TARGET} RUNTIME DESTINATION bin)
```

Comment:
Please add binaries into the export set too

Reply:
this package should export only libraries. The binaries (proof-producer executable) not used outside

Reply:
Is it a common recommendation about not exporting binaries? Yes, for now they are not used outside. But eventually we may want to add some integration tests, maybe script-like ones, so the path to the binary would be helpful. Anyway, it seems kinda strange to me that some of installed targets are exported and some are not. The main purpose of installation is to use targets outside the project, so why not extend usage scenarios if we can do it easily?

Reply:
i mean proof-producer package and proof-producer binary are different things: package includes only library and can be reused by some another targets (like assigner), binary is stand alone (as it works no in master).
If required binary - just build it.
Install package provide only library, because we may wan not to build proof-producer executable when use proof-producer-lib (for example when build assigner)

Reply:
> we may wan not to build proof-producer executable when use proof-producer-lib (for example when build assigner) 

You won't be able to avoid building anyway because project config is being generated on the installation phase when all the binaries are already built. 
>  package includes only library and can be reused by some another targets (like assigner), binary is stand alone (as it works no in master).

From my perspective, project config is a helper allowing to reuse project targets no matter what they are, binaries or libraries. But I agree that reusing libraries is much more common than binaries, so feel free to resolve.